File: sdkwork-app-sdk-python/sdkwork_app_sdk/api/file_system.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import FileSystemContentUpdateForm, FileSystemCopyForm, FileSystemFileCreateForm, FileSystemFolderCreateForm, FileSystemMoveForm, FileSystemRenameForm, PlusApiResultFileSystemContentVO, PlusApiResultFileSystemDiskVO, PlusApiResultFileSystemNodeVO, PlusApiResultListFileSystemDiskVO, PlusApiResultPageFileSystemNodeVO, PlusApiResultVoid

class FileSystemApi:
    """file_system API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def rename_node(self, nodeId: str, body: FileSystemRenameForm) -> PlusApiResultFileSystemNodeVO:
        """Rename node"""
        return self._client.put(f"/app/v3/api/filesystem/nodes/{nodeId}/rename", json=body)

    def move_node(self, nodeId: str, body: FileSystemMoveForm) -> PlusApiResultFileSystemNodeVO:
        """Move node"""
        return self._client.put(f"/app/v3/api/filesystem/nodes/{nodeId}/move", json=body)

    def get_file_content(self, fileId: str) -> PlusApiResultFileSystemContentVO:
        """Get file content"""
        return self._client.get(f"/app/v3/api/filesystem/files/{fileId}/content")

    def update_file_content(self, fileId: str, body: FileSystemContentUpdateForm) -> PlusApiResultFileSystemContentVO:
        """Update file content"""
        return self._client.put(f"/app/v3/api/filesystem/files/{fileId}/content", json=body)

    def copy_node(self, nodeId: str, body: FileSystemCopyForm) -> PlusApiResultFileSystemNodeVO:
        """Copy node"""
        return self._client.post(f"/app/v3/api/filesystem/nodes/{nodeId}/copy", json=body)

    def create_folder(self, body: FileSystemFolderCreateForm) -> PlusApiResultFileSystemNodeVO:
        """Create folder"""
        return self._client.post(f"/app/v3/api/filesystem/folders", json=body)

    def create_file(self, body: FileSystemFileCreateForm) -> PlusApiResultFileSystemNodeVO:
        """Create file"""
        return self._client.post(f"/app/v3/api/filesystem/files", json=body)

    def list_nodes(self, params: Optional[Dict[str, Any]] = None) -> PlusApiResultPageFileSystemNodeVO:
        """List nodes"""
        return self._client.get(f"/app/v3/api/filesystem/nodes", params=params)

    def get_node_detail(self, nodeId: str) -> PlusApiResultFileSystemNodeVO:
        """Get node detail"""
        return self._client.get(f"/app/v3/api/filesystem/nodes/{nodeId}")

    def delete_node(self, nodeId: str) -> PlusApiResultVoid:
        """Delete node"""
        return self._client.delete(f"/app/v3/api/filesystem/nodes/{nodeId}")

    def list_disks(self) -> PlusApiResultListFileSystemDiskVO:
        """List disks"""
        return self._client.get(f"/app/v3/api/filesystem/disks")

    def get_primary_disk(self) -> PlusApiResultFileSystemDiskVO:
        """Get primary disk"""
        return self._client.get(f"/app/v3/api/filesystem/disks/default")

File: sdkwork-app-sdk-flutter/lib/src/api/file_system.dart
import '../http/client.dart';
import '../models.dart';

class FileSystemApi {
  final HttpClient _client;
  
  FileSystemApi(this._client);

  /// Rename node
  Future<PlusApiResultFileSystemNodeVO?> renameNode(String nodeId, FileSystemRenameForm body) async {
    final response = await _client.put(ApiPaths.appPath('/filesystem/nodes/${nodeId}/rename'), body: body, contentType: 'application/json');
    return response is PlusApiResultFileSystemNodeVO ? response : null;
  }

  /// Move node
  Future<PlusApiResultFileSystemNodeVO?> moveNode(String nodeId, FileSystemMoveForm body) async {
    final response = await _client.put(ApiPaths.appPath('/filesystem/nodes/${nodeId}/move'), body: body, contentType: 'application/json');
    return response is PlusApiResultFileSystemNodeVO ? response : null;
  }

  /// Get file content
  Future<PlusApiResultFileSystemContentVO?> getFileContent(String fileId) async {
    final response = await _client.get(ApiPaths.appPath('/filesystem/files/${fileId}/content'));
    return response is PlusApiResultFileSystemContentVO ? response : null;
  }

  /// Update file content
  Future<PlusApiResultFileSystemContentVO?> updateFileContent(String fileId, FileSystemContentUpdateForm body) async {
    final response = await _client.put(ApiPaths.appPath('/filesystem/files/${fileId}/content'), body: body, contentType: 'application/json');
    return response is PlusApiResultFileSystemContentVO ? response : null;
  }

  /// Copy node
  Future<PlusApiResultFileSystemNodeVO?> copyNode(String nodeId, FileSystemCopyForm body) async {
    final response = await _client.post(ApiPaths.appPath('/filesystem/nodes/${nodeId}/copy'), body: body, contentType: 'application/json');
    return response is PlusApiResultFileSystemNodeVO ? response : null;
  }

  /// Create folder
  Future<PlusApiResultFileSystemNodeVO?> createFolder(FileSystemFolderCreateForm body) async {
    final response = await _client.post(ApiPaths.appPath('/filesystem/folders'), body: body, contentType: 'application/json');
    return response is PlusApiResultFileSystemNodeVO ? response : null;
  }

  /// Create file
  Future<PlusApiResultFileSystemNodeVO?> createFile(FileSystemFileCreateForm body) async {
    final response = await _client.post(ApiPaths.appPath('/filesystem/files'), body: body, contentType: 'application/json');
    return response is PlusApiResultFileSystemNodeVO ? response : null;
  }

  /// List nodes
  Future<PlusApiResultPageFileSystemNodeVO?> listNodes(Map<String, dynamic>? params) async {
    final response = await _client.get(ApiPaths.appPath('/filesystem/nodes'), params: params);
    return response is PlusApiResultPageFileSystemNodeVO ? response : null;
  }

  /// Get node detail
  Future<PlusApiResultFileSystemNodeVO?> getNodeDetail(String nodeId) async {
    final response = await _client.get(ApiPaths.appPath('/filesystem/nodes/${nodeId}'));
    return response is PlusApiResultFileSystemNodeVO ? response : null;
  }

  /// Delete node
  Future<PlusApiResultVoid?> deleteNode(String nodeId) async {
    final response = await _client.delete(ApiPaths.appPath('/filesystem/nodes/${nodeId}'));
    return response is PlusApiResultVoid ? response : null;
  }

  /// List disks
  Future<PlusApiResultListFileSystemDiskVO?> listDisks() async {
    final response = await _client.get(ApiPaths.appPath('/filesystem/disks'));
    return response is PlusApiResultListFileSystemDiskVO ? response : null;
  }

  /// Get primary disk
  Future<PlusApiResultFileSystemDiskVO?> getPrimaryDisk() async {
    final response = await _client.get(ApiPaths.appPath('/filesystem/disks/default'));
    return response is PlusApiResultFileSystemDiskVO ? response : null;
  }
}
